### PR TITLE
Tsodyks synapse model with common properties

### PR DESCRIPTION
**Pull request by suku248**
- homogeneous synaptic weight
- common properties U, tau_psc, tau_fac, tau_rec
- tests now check both tsodyks_synapse and tsodyks_synapse_hom

I suggest @heplesser @mhelias as code reviewers.

### CODE REVIEW

File: models/tsodyks_connection_hom.h [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef TSODYKS_CONNECTION_HOM_H
+#define TSODYKS_CONNECTION_HOM_H
+
+
+/* BeginDocumentation
+  Name: tsodyks_synapse - Synapse type with short term plasticity using homogeneous
```
Comment (heplesser) [inline models/tsodyks_connection_hom.h]: This should be `tsodyks_synapse_hom`,  shouldn't it? The body of the documentation should then also point out which parameters are shared, and contain a link to `tsodyks_synapse`.

File: models/tsodyks_connection_hom.h [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+  // this has no influence on the dynamics, IF y = z = 0 initially
+  // !!! x != 1.0 -> z != 0.0 -> t_lastspike_=0 has influence on dynamics
+
+  // propagator
+  // TODO: use expm1 here instead, where applicable
+  double_t Puu = ( cp.tau_fac_ == 0.0 ) ? 0.0 : std::exp( -h / cp.tau_fac_ );
+  double_t Pyy = std::exp( -h / cp.tau_psc_ );
+  double_t Pzz = std::exp( -h / cp.tau_rec_ );
+
+  // double_t Pzy = (Pyy - Pzz) * tau_rec_ / (tau_psc_ - tau_rec_);
```
Comment (heplesser) [inline models/tsodyks_connection_hom.h]: Could you remove this dead code? It should also be removed from the original tsodyks-synapse code.

File: models/tsodyks_connection_hom.h [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+  double_t Pxz = 1.0 - Pzz;
+
+  double_t z = 1.0 - x_ - y_;
+
+  // propagation t_lastspike -> t_spike
+  // don't change the order !
+
+  u_ *= Puu;
+  x_ += Pxy * y_ + Pxz * z;
+  // z = Pzz * z_ + Pzy * y_;
```
Comment (heplesser) [inline models/tsodyks_connection_hom.h]: This line should also go.

File: models/tsodyks_connection_hom.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+void
+TsodyksHomCommonProperties::set_status( const DictionaryDatum& d, ConnectorModel& cm )
+{
+  CommonPropertiesHomW::set_status( d, cm );
+
+  updateValue< double_t >( d, "U", U_ );
+  updateValue< double_t >( d, "tau_psc", tau_psc_ );
+  updateValue< double_t >( d, "tau_rec", tau_rec_ );
+  updateValue< double_t >( d, "tau_fac", tau_fac_ );
+}
```
Comment (heplesser) [inline models/tsodyks_connection_hom.cpp]: Shouldn't we check for legal values here, i.e., tau_* > 0 and 0 <= U <= 1? These checks are also missing in the original and in tsodyks2_synapse.

**Comment by heplesser** [comment]
@suku248 I think the code is mostly fine, but the online documentation needs to be adjusted, and you should add missing checks on the parameters, as noted above.

File: models/tsodyks_connection_hom.h [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+     y         double - initial fraction of synaptic vesicles in the synaptic cleft [0,1]
+
+  References:
+   [1] Tsodyks, Uziel, Markram (2000) Synchrony Generation in Recurrent Networks
+       with Frequency-Dependent Synapses. Journal of Neuroscience, vol 20 RC50
+
+  Transmits: SpikeEvent
+
+  FirstVersion: March 2006
+  Author: Moritz Helias, Susanne Kunkel
```
Comment (mhelias) [inline models/tsodyks_connection_hom.h]: The author line should mention the author of this homogeneous version first.

Comment (mhelias) [inline models/tsodyks_connection_hom.h]: I did not find any further issues than the ones pointed out by HEP apart from the authorship, as stated above.

**Comment by heplesser** [comment]
:+1: for merge provided checks pass again.

**Comment by jougs** [comment]
Thanks for your work on this. I'm merging as the tests pass and there's no further issues.

**Pull request merged**